applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_hydraulic_solver.py

```diff
@@ -0,0 +1,692 @@
+            computing_model_part,
+            computing_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE])
+
+        # Finding nodal and elemental neighbors
+        data_communicator = computing_model_part.GetCommunicator().GetDataCommunicator()
+        neighbour_search = KratosMultiphysics.FindGlobalNodalNeighboursProcess(
+            data_communicator,
+            computing_model_part)
+        neighbour_search.Execute()
+
+        elemental_neighbour_search = KratosMultiphysics.GenericFindElementalNeighboursProcess(
+            computing_model_part)
+        elemental_neighbour_search.Execute()
+
+        # Set and initialize the solution strategy
+        solution_strategy = self._GetSolutionStrategy()
+        solution_strategy.SetEchoLevel(self.settings["echo_level"].GetInt())
+        solution_strategy.Initialize()
+
+        # Set nodal properties after setting distance(level-set).
+        self._SetNodalProperties()
+
+        # Initialize the distance correction process
+        self._GetDistanceModificationProcess().ExecuteInitialize()
+        self._GetDistanceModificationProcess().ExecuteInitializeSolutionStep()
+
+        # Instantiate the level set convection process
+        # Note that is is required to do this in here in order to validate the defaults and set the corresponding distance gradient flag
+        # Note that the nodal gradient of the distance is required either for the eulerian BFECC limiter or by the algebraic element antidiffusivity
+
+        self._GetLevelSetConvectionProcess()
+
+        # Instantiate the eulerian historical data convection
+        if self.eulerian_fm_ale:
+            self._GetEulerianFmAleProcess()
+
+        self.mass_source = False
+        if self.settings["formulation"].Has("mass_source"):
+            self.mass_source = self.settings["formulation"]["mass_source"].GetBool()
+
+        # Non historical variable are initilized in order to avoid memory problems
+
+        KratosMultiphysics.VariableUtils().SetNonHistoricalVariableToZero(KratosMultiphysics.ACCELERATION, self.main_model_part.Nodes)
+        if self.artificial_viscosity:
+            KratosMultiphysics.VariableUtils().SetNonHistoricalVariableToZero(KratosCFD.ARTIFICIAL_DYNAMIC_VISCOSITY, self.main_model_part.Elements)
+
+        #FIXME: Check that this is required. Note that right now the convection process is the one computing NODAL_H if nodal tau is active
+        find_nodal_h = KratosMultiphysics.FindNodalHNonHistoricalProcess(self.GetComputingModelPart())
+        find_nodal_h.Execute()
+
+        KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Solver initialization finished.")
+
+
+    def InitializeSolutionStep(self):
+
+        # Inlet and outlet water discharge is calculated for current time step, first discharge and the considering the time step inlet and outlet volume is calculated
+        if self.mass_source:
+            self._ComputeInitialStepWaterVolume()
```

Comment:
```suggestion
            self._ComputeStepInitialWaterVolume()
```
Very minor, but InitialStep would refer to the first step. You mean current step initial volume.